src/OAuthClient/APIClient.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace NYPL\HoldRequestResultConsumer\OAuthClient;
+
+use GuzzleHttp\Client;
+use NYPL\HoldRequestResultConsumer\Config\Config;
+
+abstract class APIClient
+{
+    const CLIENT_TIMEOUT = 10;
+
+    /**
+     * @var Client
+     */
+    protected static $client;
+
+    /**
+     * @param array $options
+     *
+     * @return array
+     */
+    protected static function getOptions(array $options = [])
+    {
+        $options['headers']['Authorization'] = 'Bearer ' . OAuthClient::getAccessToken();
+
+        return $options;
+    }
+
+    /**
+     * @return Client
+     */
+    protected static function getClient()
+    {
+        if (!self::$client) {
+            self::initializeClient();
+        }
+
+        return self::$client;
+    }
+
+    /**
+     * @param Client $client
+     */
+    protected static function setClient($client)
+    {
+        self::$client = $client;
+    }
+
+    protected static function initializeClient()
+    {
+        self::setClient(new Client([
+            'base_uri' => Config::API_BASE_URL,
+            'timeout'  => self::CLIENT_TIMEOUT,
+        ]));
+    }
+
+    /**
+     * @param string $uri
+     * @param array $options
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    public static function get($uri = '', array $options = [])
+    {
+        return self::getClient()->get(
+            $uri,
+            self::getOptions($options)
+        );
+    }
+
+    /**
+     * @param string $uri
+     * @param array $options
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    public static function post($uri = '', array $options = [])
+    {
+        return self::getClient()->post(
+            $uri,
+            self::getOptions($options)
+        );
+    }
+
+    /**
+     * @param string $uri
+     * @param array $options
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    public static function put($uri = '', array $options = [])
+    {
+        return self::getClient()->put(
+            $uri,
+            self::getOptions($options)
+        );
+    }
+
+    /**
+     * @param string $uri
+     * @param array $options
+     *
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    public static function patch($uri = '', array $options = [])
+    {
+        return self::getClient()->patch(
+            $uri,
+            self::getOptions($options)
+        );
+    }
+}
```

Comment:
Changes seem good to me. Are you publishing to an email service or actually sending emails?